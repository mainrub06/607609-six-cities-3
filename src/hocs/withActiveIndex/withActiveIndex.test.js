import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withActiveIndex from "./withActiveIndex.jsx";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
};

const MockComponentWrapped = withActiveIndex(MockComponent);

it(`Render HOC withAudioIndex with mock Component`, () => {
  const tree = renderer
      .create((
        <MockComponentWrapped activeItemIndex = {0}/>
      ), {
        createNodeMock() {
          return {};
        }
      })
      .toJSON();

  expect(tree).toMatchSnapshot();
});
