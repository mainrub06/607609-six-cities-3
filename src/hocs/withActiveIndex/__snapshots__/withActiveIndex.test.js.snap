// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HOC withAudioIndex with mock Component 1`] = `<div />`;
